zcash_client_backend/proto/proposal.proto

```diff
@@ -5,34 +5,41 @@
 syntax = "proto3";
 package cash.z.wallet.sdk.ffi;
 
-// A data structure that describes the inputs to be consumed and outputs to
-// be produced in a proposed transaction.
+// A data structure that describes the a series of transactions to be created.
 message Proposal {
+    // The version of this serialization format.
     uint32 protoVersion = 1;
+    // The fee rule used in constructing this proposal
+    FeeRule feeRule = 2;
+    // The target height for which the proposal was constructed
+    //
+    // The chain must contain at least this many blocks in order for the proposal to
+    // be executed.
+    uint32 minTargetHeight = 3;
+    // The series of transactions to be created.
+    repeated ProposalStep steps = 4;
```

Comment:
The validity of a series of proposal steps will now be checked via the inter-step references. Since different wallets may support different subsets of inter-step references (for example, `zcash_client_backend` will support only transparent inter-step references after this PR) I don't think it makes sense to add restrictions at the encoding layer.